### PR TITLE
[device]: Add new CIG device CS6436-54P and CS5435-54P, also update code for CS…

### PR DESCRIPTION
**Pull request by michealylj1**
New platform of CIG CS6436-54P and CS5435-54P device:
ASIC Vendor: Nephos
Swich ASIC: NP8365
Port Configuration: 25GX48,100GX6

New platform of CIG CS5435-54P device:
ASIC Vendor: Nephos
Swich ASIC: NP8363
Port Configuration: 10GX48,100GX6

Update device and platform files for CIG CS6436-56P device
ASIC Vendor: Nephos
Swich ASIC: NP8366
Port Configuration: 25GX48,100GX8

SONiC Image: sonic-nephos.bin
- What I did
- How to verify it
sensors
show interfaces status
show interfaces transceiver presence
show interfaces transceiver eeprom
show platform syseeprom
show platform psustatus

[cig_devices_sonic_CLI.txt](https://github.com/Azure/sonic-buildimage/files/4084680/cig_devices_sonic_CLI.txt)

Signed-off-by: yanglijie@cigtech.com

### CODE REVIEW

File: device/cig/x86_64-cig_cs5435_54p-r0/Cig-CS5435-54P/port_config.ini [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# name          lanes             alias             index          speed
+Ethernet0       8                 Ethernet1/1       0              10000
```
Comment (yxieca) [inline device/cig/x86_64-cig_cs5435_54p-r0/Cig-CS5435-54P/port_config.ini]: Normally, we recommend having 'index' match up with front panel port numbering.

Most devices' front panel ports start from number 1. Is this device port number starting from 0?

File: device/cig/x86_64-cig_cs6436_54p-r0/Cig-CS6436-54P/port_config.ini [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# name          lanes             alias             index          speed
+Ethernet0       8                 Ethernet1/1       0              25000
```
Comment (yxieca) [inline device/cig/x86_64-cig_cs6436_54p-r0/Cig-CS6436-54P/port_config.ini]: Same question as above. Is front panel port start with 0?

File: device/cig/x86_64-cig_cs5435_54p-r0/Cig-CS5435-54P/port_config.ini (hunk 1 above)
Comment (michealylj1) [reply device/cig/x86_64-cig_cs5435_54p-r0/Cig-CS5435-54P/port_config.ini]: yes, the front panel ports  of our devices start from 1, I just update the file.

File: device/cig/x86_64-cig_cs6436_54p-r0/Cig-CS6436-54P/port_config.ini (hunk 2 above)
Comment (michealylj1) [reply device/cig/x86_64-cig_cs6436_54p-r0/Cig-CS6436-54P/port_config.ini]: Updated

**Comment by michealylj1** [comment]
Hi Xieying,

I just update the port config file and then upload it into our device, It runs well.

[cig_devices_sonic_CLI_0211.txt](https://github.com/Azure/sonic-buildimage/files/4190748/cig_devices_sonic_CLI_0211.txt)

Please help me merge the FR into the 201911 and master branch.

Thanks,
Michael

**Comment by yxieca** [comment]
@michealylj1 Sorry you took my comments in the wrong way.

2 things:

1. I didn't ask you to change the name (1st) column Ethernet<x>. I was asking about the index (4th) column.

2. You shouldn't create PR directly against 201811 branch. You should always create PR against master branch and let me know where you want the change to propagate. I don't think we should take this change in 201811 unless there is a good reason. But if you want the change in 201911. Let me know and I can help you put the flag on.

Cheers,
Ying

**Comment by michealylj1** [comment]
Hi Ying,

Sorry mistake your comments.
I just update the index column in the port config files and sfputil python files, Please check it.
As our several customers are still working on the 201811 branch using the CIG devices, we have to merge the PR into 201811 branch.

I will create new PR for master and 201911 branch.

Thanks,
Michael

**Comment by yxieca** [comment]
@michealylj1 it is ok. Your current change looks good and passing tests. It is easier to create PR against master branch and cherry-pick to all feature branches as needed.

Do you foresee master branch change being different from this change? If so, then we can merge this one. Otherwise, I prefer having one PR and cherry-pick.

**Comment by michealylj1** [comment]
Hi Ying,

It's almost same. Please help me do cherry-pick to the master and 201911 branch.

Thanks,
Michael

**Comment by yxieca** [comment]
@michealylj1 Please create a master branch PR. I'll cherry-pick that change into 201811 and 201911. Thanks. We don't cherry-pick from feature branch to master branch. You can leave this PR open for the time being. In case the master branch change cannot be cherry-picked cleanly. Then we will use this PR for 201811.

**Comment by michealylj1** [comment]
Hi Ying,

I've created new PR [4157] for master branch.  Please help me merge it.
As the kernel version in the master branch is different with the 201811 and 201911 branch, you can use the PR for 201811 and 201911 branch.

Thanks,
Michael

**Pull request merged**